lib/cli/collect-files.js

```diff
@@ -44,14 +45,34 @@ module.exports = ({
       return [...specFiles, ...moreSpecFiles];
     } catch (err) {
       if (err.code === NO_FILES_MATCH_PATTERN) {
-        unmatched.push({message: err.message, pattern: err.pattern});
+        unmatchedSpecFiles.push({message: err.message, pattern: err.pattern});
         return specFiles;
       }
 
       throw err;
     }
   }, []);
 
+  // check that each file passed in to --file exists
+
+  const unmatchedFiles = [];
+  fileArgs.forEach(file => {
+    const fileAbsolutePath = path.resolve(file);
+    try {
+      require.resolve(fileAbsolutePath);
```

Comment:
I would suggest this comment to explain the difference to https://github.com/mochajs/mocha/blob/2f3fedcc41cbb9d3e503d84098fcc07d7c3c49f1/lib/cli/lookup-files.js#L76-L77

```suggestion
      // Used instead of fs.existsSync to ensure that file-ending less files are still resolved correctly
      require.resolve(fileAbsolutePath);
```

Reply:
For the docs: I removed the `directory|glob` portion so that it only shows `--file <file>` now

- not sure if you were thinking of adding anything else so I only removed the unsupported file arguments

For the comment in `mocha/lib/cli/collect-files.js`: I added the comment you suggested